tensorly/solvers/proximal.py

```diff
@@ -0,0 +1,860 @@
+    proximal operator.
+
+    Parameters
+    ----------
+    non_negative : bool or dictionary
+        This constraint is clipping negative values to '0'.
+        If it is True, non-negative constraint is applied to all modes.
+    l1_reg : float or list or dictionary, optional
+        Penalizes the factor with the l1 norm using the input value as regularization parameter.
+    l2_reg : float or list or dictionary, optional
+        Penalizes the factor with the l2 norm using the input value as regularization parameter.
+    l2_square_reg : float or list or dictionary, optional
+        Penalizes the factor with the l2 square norm using the input value as regularization parameter.
+    unimodality : bool or dictionary, optional
+        If it is True, unimodality constraint is applied to all modes.
+        Applied to each column seperately.
+    normalize : bool or dictionary, optional
+        This constraint divides all the values by maximum value of the input array.
+        If it is True, normalize constraint is applied to all modes.
+    simplex : float or list or dictionary, optional
+        Projects on the simplex with the given parameter
+        Applied to each column seperately.
+    normalized_sparsity : float or list or dictionary, optional
+        Normalizes with the norm after hard thresholding
+    soft_sparsity : float or list or dictionary, optional
+        Impose that the columns of factors have L1 norm bounded by a user-defined threshold.
+    smoothness : float or list or dictionary, optional
+        Optimizes the factors by solving a banded system
+    monotonicity : bool or dictionary, optional
+        Projects columns to monotonically decreasing distrbution
+        Applied to each column seperately.
+        If it is True, monotonicity constraint is applied to all modes.
+    hard_sparsity : float or list or dictionary, optional
+        Hard thresholding with the given threshold
+    n_const : int
+        Number of constraints. If it is None, function returns input tensor.
+        Default : 1
+    order : int
+        Specifies which constraint to implement if several constraints are selected as input
+        Default : 0
+    Returns
+    -------
+    constraint : string
+    parameter : float
+    """
+    constraints = [None] * n_const
+    parameters = [None] * n_const
+    if non_negative:
+        if isinstance(non_negative, dict):
+            modes = list(non_negative)
+            for i in range(len(modes)):
+                constraints[modes[i]] = "non_negative"
+        else:
+            for i in range(len(constraints)):
+                constraints[i] = "non_negative"
+    if l1_reg:
+        if isinstance(l1_reg, dict):
+            modes = list(l1_reg)
+            for i in range(len(modes)):
+                if constraints[modes[i]] is not None:
+                    raise ValueError(
+                        "You selected two constraints for the same mode. Consider to check your input"
```

Comment:
Can the two constraints for the same mode check be done in a constraint-general way? In other words, does it have to be repeated within each if-else block? It might be more readable if it can be moved out.